src/ext/env_config.c

```diff
@@ -1,17 +1,32 @@
 #include <SAPI.h>
 #include <Zend/zend_types.h>
 
+#include "coms_curl.h"
 #include "env_config.h"
 
-#define EQUALS(stra, strb) (memcmp(stra, strb, sizeof(strb) - 1) == 0)
+#define EQUALS(stra, stra_len, literal_strb) \
+    (stra_len == (sizeof(literal_strb) - 1) && memcmp(stra, literal_strb, sizeof(literal_strb) - 1) == 0)
 
 char *get_local_env_or_sapi_env(char *name) {
-    TSRMLS_FETCH();
     char *env = NULL, *tmp = getenv(name);
     if (tmp) {
-        env = estrdup(tmp);
+        env = ddtrace_strdup(tmp);
     } else {
+        // reading sapi_getenv from within writer thread can and will lead to undefined behaviour
+        if (ddtrace_in_writer_thread()) {
+            return NULL;
+        }
+
+        TSRMLS_FETCH();
```

Comment:
This macro adds quite a bit of overhead which might be one of the the reasons why config has been a bit slow. I know this would make the API's really ugly, but perhaps we should consider adding the `TSRMLS_DC` macros to the these API's. What are your thoughts on this? :)

Reply:
👍 Agreed, I'll tweak the api, also this is not thread safe, so I'll probably make this a mandatory parameters, and cause the background thread to only allow read only access to config.